fhir-search/src/test/java/com/ibm/fhir/search/test/InclusionParameterParseTest.java

```diff
@@ -327,4 +340,151 @@ public void testMultiIncludeRevinclude() throws Exception {
+
+        assertNotNull(searchContext);
+        assertTrue(searchContext.hasIncludeParameters());
+        assertEquals(expectedIncludeParms.size(), searchContext.getIncludeParameters().size());
+        for (InclusionParameter includeParm : expectedIncludeParms) {
+            assertTrue(expectedIncludeParms.contains(includeParm));
+        }
+
+        assertFalse(searchContext.hasRevIncludeParameters());
+
+        String selfUri = SearchUtil.buildSearchSelfUri("http://example.com/Patient", searchContext);
+        assertTrue(selfUri.contains(include1));
+        assertTrue(selfUri.contains(include2));
+        assertTrue(selfUri.contains(include3));
+        assertTrue(selfUri.contains(include4));
+        assertTrue(selfUri.contains(include5));
+        assertTrue(selfUri.contains(include6));
+    }
+
+    @Test
+    public void testWildcardRevIncludeNoMatchingTargetType() throws Exception {
+        Map<String, List<String>> queryParameters = new HashMap<>();
+        Class<Condition> resourceType = Condition.class;
+
+        queryParameters.put("_revinclude", Collections.singletonList("Patient:*:Condition"));
+        FHIRSearchContext searchContext = SearchUtil.parseQueryParameters(resourceType, queryParameters);
+
+        assertNotNull(searchContext);
+        assertFalse(searchContext.hasIncludeParameters());
+        assertFalse(searchContext.hasRevIncludeParameters());
+
+        String selfUri = SearchUtil.buildSearchSelfUri("http://example.com/Condition", searchContext);
+        assertTrue(selfUri.equals("http://example.com/Condition?_count=10&_page=1"));
+    }
+
+    @Test
+    public void testWildcardRevIncludeSingleMatchingTargetType() throws Exception {
+        Map<String, List<String>> queryParameters = new HashMap<>();
+        Class<Encounter> resourceType = Encounter.class;
+
+        queryParameters.put("_revinclude", Collections.singletonList("MedicationAdministration:*:Encounter"));
+        FHIRSearchContext searchContext = SearchUtil.parseQueryParameters(resourceType, queryParameters);
+
+        assertNotNull(searchContext);
+        assertTrue(searchContext.hasRevIncludeParameters());
+        assertEquals(1, searchContext.getRevIncludeParameters().size());
+        InclusionParameter incParm = searchContext.getRevIncludeParameters().get(0);
+        assertEquals("MedicationAdministration", incParm.getJoinResourceType());
+        assertEquals("context", incParm.getSearchParameter());
+        assertEquals("Encounter", incParm.getSearchParameterTargetType());
+        assertFalse(searchContext.hasIncludeParameters());
+
+        String selfUri = SearchUtil.buildSearchSelfUri("http://example.com/Encounter", searchContext);
+        assertTrue(selfUri.equals("http://example.com/Encounter?_count=10&_revinclude=" + incParm.getJoinResourceType() +
+            ":" + incParm.getSearchParameter() + ":" + incParm.getSearchParameterTargetType() + "&_page=1"));
+    }
+
+    @Test
```

Comment:
Let's add a set of extra tests. That has two _include search parameters on Patient and Provenance.  we want to double check that it works across multiple resources.

Reply:
Added new unit tests.